### DISCUSSION TITLE
Wrong parameter passed to raw-body in api-utils.apiResolver

### DISCUSSION THREAD

**Discussion by bbourn** [vercel/next.js#11982] [Help]
# Bug report

## Describe the bug

When using api-utils.apiResolver with Jest to test my endpoints, I cannot get a json body to pass through to the handler.

## To Reproduce

Use in a serverless now / nextjs project with pages/api component jest test like this:
```
const requestHandler = async (req, res): void => {
      req.method = 'POST'
      req.body = { test: 'body' }
      req.headers = { 'content-type': 'application/json' }
      return apiResolver(req, res, undefined, handlerPagesAPIComponentToTest)
    }
```
## Expected behavior

When handlerComponentToTest gets the req, it should have a body with above values.

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of Next.js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

## Additional context

If I add lines to api-utils.parseBody function to convert req.body to a stream and pass only that to raw_body_1 call, it works as expected.

```
   let buffer;
    var Readable = require('readable-stream').Readable
    var stream = new Readable()
    stream.push(JSON.stringify(req.body))
    stream.push(null)
try {
        buffer = await raw_body_1.default(stream, { length: null, encoding, limit });
```

## Other blockers on the way here were
Had to define
```
export const config = {
  api: {
    bodyParser: true
  }
}
```

And parseBody only looks for lowercase 'content-type'
```
const contentType = content_type_1.parse(req.headers['content-type'] || 'text/plain');
```

**Comment by timneutkens** [comment]
This sounds more like a feature request to expose the api handler middlewares 🤔 seems like you're using internal files which can change at any time.

**Comment by bbourn** [comment]
I am using internal files.  And I called it a bug cause there was no way that I could see to use body without changing the parameter here.  But after I posted realized this is Zeit.  And probably the wrong place to post, but it might help anyone else that wants to use .body in their tests of next pages/api endpoints.
